sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -755,6 +755,10 @@ private[hive] class HiveClientImpl(
     client.getTablesByPattern(dbName, pattern).asScala
   }
 
+  override def listViews(dbName: String, pattern: String): Seq[String] = withHiveState {
+    shim.getTablesByType(client, dbName, pattern, HiveTableType.VIRTUAL_VIEW)
+  }
```

Comment:
Plz add tests in the hive side, too.

Reply:
Yea, updated in dc6e3cb21e5de2a2c591b42014cc03fc4418cca3. Not sure if it's enough.

Reply:
Yea and I think the added ones are good enough.